torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3039,6 +3042,9 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::allreduce_sparse(
     const AllreduceOptions& opts) {
   TORCH_CHECK(tensors.size() == 1, MULTI_DEVICE_ERROR_MSG);
   auto tensor = tensors.back();
+  TORCH_CHECK(
+      !isFloat8Type(tensor.scalar_type()),
```

Comment:
shall we also add some checks to reduce_scatter?

Reply:
ahh yeah, will add